db/indexes_test.go

```diff
@@ -57,53 +59,42 @@ func TestInitializeIndexes(t *testing.T) {
 
 			// Make sure we can drop and reinitialize twice
 			for i := 0; i < 2; i++ {
-				dropErr := base.DropAllIndexes(base.TestCtx(t), n1qlStore)
-				require.NoError(t, dropErr, "Error dropping all indexes")
-
-				initErr := InitializeIndexes(n1qlStore, test.xattrs, 0, true)
-				require.NoError(t, initErr, "Error initializing all indexes")
-
-				// Recreate the primary index required by the test bucket pooling framework
-				err := n1qlStore.CreatePrimaryIndex(base.PrimaryIndexName, nil)
-				require.NoError(t, err)
-
-				validateErr := validateAllIndexesOnline(b)
-				require.NoError(t, validateErr, "Error validating indexes online")
+				err := dropAndInitializeIndexes(base.TestCtx(t), n1qlStore, b, test.xattrs)
+				require.NoError(t, err, "Error dropping and initialising all indexes on bucket")
 			}
 		})
 	}
 
 }
 
 // Reset bucket state
-func validateAllIndexesOnline(bucket base.Bucket) error {
+func validateAllIndexesOnline(bucket base.Bucket, xattrs bool) error {
 
-	n1QLStore, ok := base.AsN1QLStore(bucket)
-	if !ok {
-		return fmt.Errorf("Bucket is not gocb bucket: %T", bucket)
-	}
+	col, err := base.AsCollection(bucket)
 
-	// Retrieve all indexes
-	getIndexesStatement := fmt.Sprintf("SELECT indexes.name, indexes.state from system:indexes where keyspace_id = %q", n1QLStore.GetName())
-	results, err := n1QLStore.Query(getIndexesStatement, nil, base.RequestPlus, true)
-	if err != nil {
-		return err
-	}
+	cluster := col.GetCluster()
+	mgr := cluster.QueryIndexes()
 
-	var indexRow struct {
-		Name  string
-		State string
+	watchOption := gocb.WatchQueryIndexOptions{
+		WatchPrimary:   true,
+		ScopeName:      "sg_test_1",
+		CollectionName: "sg_test_1",
```

Comment:
you want to do to something like:

```
collection, err := base.AsCollection(bucket)
require.NoError(err)
```

then use `ScopeName: collection.ScopeName()` and `CollectionName: collection.Name()`